src/api/mod.rs

```diff
@@ -96,3 +97,31 @@ pub(in crate::api) async fn check_community_ban(
     Ok(())
   }
 }
+
+pub(in crate::api) async fn linked_instances(pool: &DbPool) -> Result<Vec<String>, LemmyError> {
+  let mut instances: Vec<String> = Vec::new();
+
+  if Settings::get().federation.enabled {
+    let distinct_communities = blocking(pool, move |conn| {
+      Community::distinct_federated_communities(conn)
+    })
+    .await??;
+
+    for actor_id in &distinct_communities {
+      instances.push(Url::parse(actor_id)?.host_str().unwrap_or("").to_string());
```

Comment:
I couldn't figure out how to convert to stream operators, mainly due to `Url::parse...?`. It says it would need to return a result in the stream operator closure, and I couldn't figure it out.

Reply:
Do it like this:

https://github.com/LemmyNet/lemmy/blob/main/src/routes/feeds.rs#L252

Reply:
Thx, ya it did.